src/types/inp.ts

```diff
@@ -24,6 +24,129 @@ export interface INPMetric extends Metric {
+  /**
+   * The execution duration of the slowest script. Note this may be longer
+   * than the intersectingScriptDuration if the INP interaction happened
+   * mid-script.
+   */
+  executionDuration: number;
+  /**
+  /**
+   * The forced style and layoult duration of the slowest script. Note this
+   * may be longer than the intersectingScriptDuration if the INP interaction
+   * happened mid-script.
+   */
+  forcedStyleAndLayoutDuration: number;
+  /**
+   * The pause duration of the slowest script. Note this may be longer
+   * than the intersectingScriptDuration if the INP interaction happened
+   * mid-script.
+   */
+  pauseDuration: number;
+  /**
+   * The invokerType of the slowest script.
+   */
+  invokerType: ScriptInvokerType;
+  /**
+   * The invoker of the slowest script.
+   */
+  invoker?: string;
+  /**
+   * The sourceURL of the slowest script.
+   */
+  sourceURL?: string;
+  /**
+   * The sourceFunctionName of the slowest script.
+   */
+  sourceFunctionName?: string;
+  /**
+   * The sourceCharPosition of the slowest script.
+   */
+  sourceCharPosition?: number;
+}
+
+/**
+ * An object containing potentially-helpful debugging information summarized
+ * from the LongAnimationFrames intersecting the INP interaction.
+ *
+ * NOTE: Long Animation Frames below 50 milliseconds are not reported, and
+ * so their scripts cannot be included. For Long Animation Frames that are
+ * reported, only scripts above 5 milliseconds are included.
+ */
+export interface LongAnimationFrameSummary {
+  /**
+   * The number of Long Animation Frame scripts that intersect the INP
+   * interaction.
+   * NOTE: This may be be less than the total count of scripts in the Long
+   * Animation Frames as some scripts may occur before the interaction.
+   */
+  numIntersectingScripts?: number;
+  /**
+   * The number of Long Animation Frames intersecting the INP interaction.
+   */
+  numLongAnimationFrames?: number;
+  /**
+   * Summary details about the slowest Long Animation Frame script that
+   * intersects the INP interaction.
+   */
+  slowestScript?: SlowestScriptSummary;
+  /**
+   * The total blocking durations in each phase by invoker for scripts that
+   * intersect the INP interaction.
+   */
+  totalDurationsPerPhase?: Record<
+    'inputDelay' | 'processingDuration' | 'presentationDelay',
+    Record<ScriptInvokerType, number>
+  >;
+  /**
+   * The total forced style and layout durations as provided by Long Animation
+   * Frame scripts intersecting the INP interaction.
+   */
+  totalForcedStyleAndLayoutDuration?: number;
+  /**
+   * The total non-force (i.e. end-of-frame) style and layout duration from any
+   * Long Animation Frames intersecting INP interaction.
+   */
+  totalNonForcedStyleAndLayoutDuration?: number;
+  /**
+   * The total duration of Long Animation Frame scripts that intersect the INP
+   * duration. Note, this includes forced style and layout within those
+   * scripts.
+   */
+  totalScriptDuration?: number;
```

Comment:
can't some of these properties be required and not make the caller check for undefineds? e.g. shouldn't the default for this  one be 0?

Reply:
Yes they can.